pkg/clients/integration/pipelineruns.go

```diff
@@ -79,7 +79,7 @@ func (i *IntegrationController) GetBuildPipelineRun(componentName, applicationNa
 		}
 
 		if len(list.Items) > 0 {
-			pipelineRun = &list.Items[0]
+			pipelineRun = &list.Items[len(list.Items)-1]
```

Comment:
Based on some more discussion, I think `list.Items[]` doesn't contain PLR objects in a specific order so always accessing the last item from it _might_ give unexpected results in some cases.

Wdyt about sorting the PLR objects based on the `status.completionTime`?

![image](https://github.com/redhat-appstudio/e2e-tests/assets/43273420/a98bc7c3-048f-4d4c-bed6-54975c5216ef)

maybe like:
```
// Sort PipelineRuns by completionTime
sort.Slice(pipelineRunList, func(i, j int) bool {
	return pipelineRunList[i].Status.CompletionTime.Before(pipelineRunList[j].Status.CompletionTime)
})
```

Wdyt?

Reply:
you need start/creation time right? There is no guarantee that latest pipeline will end as the last

Reply:
My bad! Thanks!! Goinggg

Reply:
IIUC, the function will [re-trigger a new PLR](https://github.com/redhat-appstudio/e2e-tests/blob/96a7c40d440d86b2d4c3b95c000ed73b1d81ea4b/pkg/clients/has/components.go#L168) only if the previous one timed out or failed with some other error. So, sorting them by completion timestamp would still give us the latest one at the last index. Having said that, I'm fine with sorting them by start/creation time as well.

Reply:
I don't like relying on effects of functions :D